ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -55,41 +55,43 @@ class IbmMqCheck(AgentCheck):
 
     CHANNEL_COUNT_CHECK = 'ibm_mq.channel.count'
 
-    def check(self, instance):
-        config = IBMMQConfig(instance)
-        config.check_properly_configured()
+    def __init__(self, name, init_config, instances):
+        super(IbmMqCheck, self).__init__(name, init_config, instances)
+        self.config = IBMMQConfig(self.instance)
+        self.config.check_properly_configured()
 
+    def check(self, instance):
         if not pymqi:
             log.error("You need to install pymqi: %s", pymqiException)
             raise errors.PymqiException("You need to install pymqi: {}".format(pymqiException))
 
+        tags = self.config.tags  # Calculated property
```

Comment:
Shall we just convert `config.tags` and `config.no_channel_tags` into normal properties ? 

Example:

```python

class IBMMQConfig:

    def __init__(self, instance):

        self.tags_no_channel = [
            "queue_manager:{}".format(self.queue_manager_name),
            "mq_host:{}".format(self.host),  # 'host' is reserved and 'mq_host' is used instead
            "port:{}".format(self.port),
        ] + self.custom_tags

        self.tags = self.tags_no_channel.copy()
        self.tags.append("channel:{}".format(self.channel))
        
```